File: backend/src/main/java/ropold/backend/model/RealEstateModel.java
package ropold.backend.model;

import java.time.LocalDate;
import java.util.List;

public record RealEstateModel(
        String id,
        String realEstateTitle,
        String description,
        String address,
        double price,
        PriceType priceType,
        List<RoomModel> rooms,
        double totalFloorArea,
        double totalLivingAreaWoFlV,
        String githubId,
        LocalDate createdAt,
        String imageUrl
) {
}

File: frontend/src/components/model/RealEstateModel.ts
import type {RoomModel} from "./RoomModel.ts";
import type {PriceType} from "./PriceType.ts";

export type RealEstateModel = {
    id: string;
    realEstateTitle: string;
    description: string;
    address: string;
    price: number;
    priceType: PriceType;
    rooms: RoomModel[];
    totalFloorArea: number;
    totalLivingAreaWoFlV: number;
    githubId: string;
    createdAt: string;
    imageUrl: string | null;
};

export const DefaultRealEstate: RealEstateModel = {
    id: "",
    realEstateTitle: "",
    description: "",
    address: "",
    price: 0,
    priceType: "PURCHASE",
    rooms: [],
    totalFloorArea: 0,
    totalLivingAreaWoFlV: 0,
    githubId: "",
    createdAt: "",
    imageUrl: null
};
